scripts/remove_after_use/fix_unclaimed_records.py

```diff
@@ -0,0 +1,35 @@
+import sys
+import logging
+from website.app import setup_django
+
+setup_django()
+from osf.models import OSFUser
+logger = logging.getLogger(__name__)
+
+
+def add_missing_unclaimed_record(user, node, dry_run):
+    # Get referrer from logs
+    for log in node.logs.filter(action='contributor_added').order_by('date'):
+        if user._id in log.params['contributors']:
+            if log.user_id:
+                referrer = OSFUser.objects.get(id=log.user_id)
+                user.add_unclaimed_record(node, referrer, user.fullname)
+                if not dry_run:
+                    user.save()
+                logger.info('User {} has been given an unclaimed record with name {} for node {}'.format(user._id, user.fullname, node._id))
+            else:
+                logger.info('User {} has could not be given a record, because their referer was anonymous'.format(user._id))
+
+
+def main(dry_run=True):
+    users = OSFUser.objects.filter(date_disabled=None,
+                                   is_registered=False).exclude(nodes__type='osf.quickfiles').include(None).distinct()
+    logger.info('{} users without unclaimed records'.format(users.count()))
```

Comment:
This was the initial output I got, 
```
[__main__]  INFO: 36291 users without unclaimed records
```
which seems a bit high, but four iterations in it ran into a unicode error logging a name.

Reply:
@sloria I vaguely remember discussing that these could be spam users, not sure what else could be causing this obviously very high number.